packages/ui/src/theme/createTheme.ts

```diff
@@ -63,6 +63,35 @@ function setupTokens(obj: any, path = []) {
   return tokens;
 }
 
+/**
+ * This function takes in an array of WebDesignTokens and filters out the duplicate WebDesignTokens
+ * that are defined in the DUPLICATE_STATE_TOKENS list. This is designed to remove the duplicate tokens
+ * that were created due to inconsistent naming.  This should be removed when the tokens are removed in the
+ * next major version release.
+ * @param tokens
+ * @returns WebDesignTokens[]
+ */
+function removeDuplicateStateTokens(
+  tokens: WebDesignToken[]
+): WebDesignToken[] {
+  let tokenList = tokens;
+  DUPLICATE_STATE_TOKENS.forEach((duplicateToken) => {
+    const filteredTokens = tokenList.filter(
```

Comment:
Is this filter necessary? Shouldn't we know that what inside `DUPLICATE_STATE_TOKENS` is guaranteed to be duplicate and safe to filter out?

Reply:
The reason we have this filter instead of just removing the tokens, is that we only want to remove duplicate tokens if more than one exists in the theme.  This happens because the users are currently using the deprecated tokens and once they switch over to the `_state` tokens two tokens will be generated (because the old deprecated version will still be added through the defaultTheme) so here we are checking that both exist and if they do we remove the old one, which we only want to do if the new token is being used.

Reply:
I think we could optimize this function a bit. From what it looks like this will iterated over the entire tokenlist (~1k tokens) N times where N is the number in DUPLICATE_STATE_TOKENS, regardless of if there are duplicates or not. Instead could we try to filter the tokenlist once to check for duplicates, something like 

```javascript
tokenList.filter((token,i) => {
  if (tokenList.indexOf(token) !== i) {
    // there is a duplicate
  }
})
```

Would this be more efficient?

Reply:
Thanks, this has been updated to make a single pass instead over the design tokens to find duplicates and a single filter to remove them